eta/core/video.py

```diff
@@ -108,6 +116,686 @@ def glob_videos(dir_):
+            timestamp = self.get_timestamp(world_time=world_time)
+        if timestamp is not None:
+            frame_number = self.get_frame_number(timestamp=timestamp)
+        return self._flat(frame_number), self._flon(frame_number)
+
+    def attributes(self):
+        _attrs = [
+            "start_time", "frame_size", "frame_rate", "total_frame_count",
+            "duration", "size_bytes", "encoding_str", "gps_waypoints"
+        ]
+        # Exclue attributres that are None
+        return [a for a in _attrs if getattr(self, a) is not None]
+
+    @classmethod
+    def build_for(
+            cls, filepath, start_time=None, gps_waypoints=None):
+        '''Builds a VideoMetadata object for the given video.
+
+        Args:
+            filepath: the path to the video on disk
+            start_time: an optional datetime specifying the start time of the
+                video
+            gps_waypoints: an optional list of GPSWaypoint instances describing
+                the GPS coordinates of the video
+
+        Returns:
+            a VideoMetadata instance
+        '''
+        vsi = VideoStreamInfo.build_for(filepath)
+        return cls(
+            start_time=start_time,
+            frame_size=vsi.frame_size,
+            frame_rate=vsi.frame_rate,
+            total_frame_count=vsi.total_frame_count,
+            duration=float(vsi.get_raw_value("duration")),
+            size_bytes=os.path.getsize(filepath),
+            encoding_str=vsi.encoding_str,
+            gps_waypoints=gps_waypoints,
+        )
+
+    @classmethod
+    def from_dict(cls, d):
+        '''Constructs a VideoMetadata from a JSON dictionary.'''
+        start_time = d.get(d["start_time"], None)
+        if start_time is not None:
+            start_time = dateutil.parser.parse(start_time)
+
+        gps_waypoints = d.get("gps_waypoints", None)
+        if gps_waypoints is not None:
+            gps_waypoints = [GPSWaypoint.from_dict(g) for g in gps_waypoints]
+
+        return cls(
+            start_time=start_time,
+            frame_size=d.get("frame_size", None),
+            frame_rate=d.get("frame_rate", None),
+            total_frame_count=d.get("total_frame_count", None),
+            duration=d.get("duration", None),
+            size_bytes=d.get("size_bytes", None),
+            encoding_str=d.get("encoding_str", None),
+            gps_waypoints=gps_waypoints,
+        )
+
+    def _init_gps(self):
+        if not self.has_gps:
+            return
+        frames = [loc.frame_number for loc in self.gps_waypoints]
+        lats = [loc.latitude for loc in self.gps_waypoints]
+        lons = [loc.longitude for loc in self.gps_waypoints]
+        self._flat = self._make_interp(frames, lats)
+        self._flon = self._make_interp(frames, lons)
+
+    def _make_interp(self, x, y):
+        return spi.interp1d(
+            x, y, kind="nearest", bounds_error=False, fill_value="extrapolate")
+
+
+class VideoFrameLabels(Serializable):
```

Comment:
Labels?
VideoFrameSemantics?
VideoFrameExtractions?
VideoFrameSemanticMetadata?  
Labels seems like the wrong name to me.  It is scene-level attributes and it has objects.  It probably should add actions in some way in the future, or other things like human poses?  It is more than labels.

Reply:
So, you like `Labels`, fine.  The CV namespace is already quite the quagmire, another patch of mud won't matter much.  We know what it means.